modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -1597,14 +1595,8 @@ void cv::gimpl::GStreamingExecutor::setSource(GRunArgs &&ins)
         island_meta_info = GIslandModel::traceIslandName(op.nh, m_gim);
 #endif // OPENCV_WITH_ITT
 
-        // If Island Executable is recompiled, all its stuff including internal kernel states
-        // are recreated and re-initialized automatically.
-        // But if not, we should notify Island Executable about new started stream to let it update
-        // its internal variables.
-        if (!islandsRecompiled)
-        {
-            op.isl_exec->handleNewStream();
-        }
+        // Notify island executable about a new stream to let it update its internal variables.
+        op.isl_exec->handleNewStream();
 
         m_threads.emplace_back(islandActorThread,
                                op.in_objects,
```

Comment:
If `islandActorThread` is infinitive actor thread (`while(true)`) then it is good idea to improve it

```
islandActorThread ()
...
GAPI_LOG_INFO(..., <name of island executor> " Started") 
while(true) {
}
GAPI_LOG_INFO(...,  Finished") 
```